src/values.cpp

```diff
@@ -343,6 +343,23 @@ auto Origin::raw() -> Type& { return this->origin; }
         },
         this->origin);
 }
+void Origin::set_file(std::optional<std::shared_ptr<std::string>> file) {
+    std::visit(
+        overloaded{
+            [&file](BinaryOrigin& origin) {
+                if (origin.location) {
+                    origin.location->file = file;
+                }
+            },
+            [&file](UnaryOrigin& origin) {
+                if (origin.location) {
+                    origin.location->file = file;
+                }
+            },
+            [&file](LiteralOrigin& origin) { origin.location.file = file; },
+            [](auto& /*unused*/) {}},
+        this->origin);
+}
```

Comment:
Should `MultipleArgsOrigin` be added once my PR with the math-functions is merged?

Reply:
Yes

Reply:
I separated the visit callbacks so this will fail to compile when we add MultipleArgsOrigin.